source/brailleDisplayDrivers/albatross.py

```diff
@@ -0,0 +1,437 @@
+		else:
+			raise RuntimeError("No Albatross found")
+
+	def terminate(self):
+		try:
+			super().terminate()
+			if self._rt:
+				self._rt.stop()
+				self._rt = None
+			# Possibly already closed.
+			if self._dev:
+				self._dev.close()
+		finally:
+			self._dev = None
+			self.numCells = 0
+
+	def _chkPort(self, port: bytes) -> bool:
+		try:
+			self._dev = hwIo.Serial(
+				port, baudrate=BAUD_RATE, stopbits=serial.STOPBITS_ONE, parity=serial.PARITY_NONE,
+				timeout=TIMEOUT, writeTimeout=WRITE_TIMEOUT, onReceive=self._onReceive)
+		except EnvironmentError:
+			log.debugWarning("", exc_info=True)
+			return False
+		return True
+
+	# Whole display should be updated at next time.
+	def _clearOldCells(self):
+		for i in range(len(self._oldCells)):
+			self._oldCells[i] = 0
+
+	# All write operations are done here.
+	def _sendToDisplay(self, data: bytes) -> bool:
+		try:
+			self._dev.write(data)
+		except serial.serialutil.SerialException:
+			log.debug("Data write failed %r" % data)
+			# Suitable initial values for reconnection.
+			# Connection worked before failure.
+			if self.numCells:
+				self.numCells = 0
+				self._clearOldCells()
+			self._dev = None
+			# Maybe display was unplugged/powered off which causes USB serial port disappearing.
+			if not self._chkPort(self._currentPort):
+				self._tryReconnect = True
+				log.debug("Port %r disappeared, trying reconnect" % self._currentPort)
+			return False
+		# Reset timer to avoid sending reduntant BOTH_BYTES packets.
+		self._rt.stop()
+		self._rt.start()
+		# Confirmation packet sent so that device stopped sending initial packet.
+		if data == ESTABLISHED:
+			# Actually input buffer should be reseted, but poor man's solution now.
+			try:
+				while len(self._dev.read(INPUT_BUF_SIZE)):
+					pass
+			except serial.serialutil.SerialException:
+				log.debug("Port %r input buffer read failed" % self._currentPort)
+				return False
+		return True
+
+	def _keepConnected(self):
+		# Can disappeared port be found again?
+		if self._tryReconnect:
+			if not self._chkPort(self._currentPort):
+				return
+			else:
+				# Port found.
+				self._tryReconnect = False
+				return
+		# Display needs this to keep connection.
+		self._sendToDisplay(BOTH_BYTES)
+
+	def _onReceive(self, data: bytes):
+		if not self._checkConnection(data):
+			return
+		else:
+			# Connected, but user exited internal menu of display.
+			if not self._exitInternalMenu(data):
+				return
+		# If Ctrl-key is pressed, then there is at least one byte to read;
+		# in single ctrl-key presses and key combinations the first key is resent as last one.
+		if ord(data) in CONTROL_KEY_CODES:
+			# at most 4 keys.
+			try:
+				data += self._dev.read(4)
+			except serial.serialutil.SerialException:
+				log.debug("Ctrl key read failed")
+				return
+		pressedKeys = set(data)
+		try:
+			inputCore.manager.executeGesture(InputGestureKeys(pressedKeys))
+		except (inputCore.NoInputGestureAction, AttributeError):
+			pass

```

Comment:
I know this is common, but swallowing the exception is bad practice. Please log at least a debug message. The alternative is a comment about why this is not helpful to log (I.E. occurs very often and provides little value).